earth_enterprise/src/SConstruct

```diff
@@ -367,6 +367,40 @@ installdirs = {
     'htdocs': InstServerDir('gehttpd/htdocs'),
 }
 
+def get_cc_version(cc_path):
+  """Extracts the version number of a C/C++ compiler (like GCC) as a list of
+  strings."""
+
+  process = subprocess.Popen(
+    [cc_path, '--version'], stdout=subprocess.PIPE)
+  (stdout, stderr) = process.communicate()
+  match = re.search('[0-9][0-9.]*', stdout)
+  # If match is None we don't know the version.
+  if match is None:
+    return None
+  version = match.group().split('.')
+
+  return version
+
+def is_version_ge(version_components, comparand_components):
+  """Tests if a version number is greater than, or equal to another version
+  number.  Both version numbers are specified as lists of version components.
+  """
+
+  if version_components is None:
```

Comment:
This will fail if comparand_components is none.

Reply:
Yes.  It also doesn't make much sense to try comparing a version to something that you don't specify.

Reply:
I think that's fine.  A caller shouldn't be asking to compare a version to something the caller is not specified.